### PR TITLE
PR to integrate JWT on login

### PR DESCRIPTION
**Pull request by iAmao**
### What does this PR do?

This PR contains completed output for generating a token for the user when user logs in successfully.

### Description of Task to be completed?

Task generates token for the user on a successful sign in. The token is signed with the user's name and id. The token is then used  to authenticate users across routes that require authentication.

### How should this be manually tested?
- Open up the terminal and type the following commands:
```
git clone https://github.com/andela/izanami-cfh.git
cd izanami-cfh
npm install
npm start
```
- Navigate to http://localhost:3000

Any background context you want to provide?

N/A

### What are the relevant pivotal tracker stories?

Users should receive a JWT on successful login

### Screenshots
N/A

### CODE REVIEW

**Comment by iAmao** [comment]
@andela-ooduye  PR updated with changes. User gets redirected with an error when they provide wrong credentials. login endpoint has been changed

**Review by ooduye — COMMENTED** [review]
Check through the HoundCI flags and correct them accordingly, please

File: public/js/controllers/index.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 angular.module('mean.system')
-.controller('IndexController', ['$scope', 'Global', '$location', 'socket', 'game', 'AvatarService', function ($scope, Global, $location, socket, game, AvatarService) {
+.controller('IndexController', ['$scope','Global','$location', 'socket', 'game', 'AvatarService', function ($scope, Global, $location, socket, game, AvatarService) {
```
[the remark this answers is not in the document]
Comment (ooduye) [reply public/js/controllers/index.js]: @andela-iamao Include the space required

File: config/routes.js [hunk 2]
```diff
@@ -1,93 +1,105 @@
...
-    app.get('/avatars', avatars.allJSON);
+  const avatars = require('../app/controllers/avatars');
+  app.get('/avatars', avatars.allJSON);
 
-    //Home route
-    var index = require('../app/controllers/index');
-    app.get('/play', index.play);
-    app.get('/', index.render);
+    // Home route
+  const index = require('../app/controllers/index');
```
[the remark this answers is not in the document]
Comment (ooduye) [reply config/routes.js:104]: Any specific reason why the packages are required within the code as opposed to it all being done at the beginning of the file?! If so, please leave a comment as to why.

File: app/views/includes/foot.jade [hunk 3]
```diff
@@ -10,20 +10,24 @@ script(type='text/javascript', src='/lib/underscore/underscore-min.js')
...
 script(type='text/javascript', src='/lib/angular-ui-utils/modules/route/route.js')
 
+
 //Application Init
 script(type='text/javascript', src='/js/init.js')
 script(type='text/javascript', src='/js/app.js')
 script(type='text/javascript', src='/js/directives.js')
 script(type='text/javascript', src='/js/filters.js')
 
+
```
Comment (ooduye) [inline app/views/includes/foot.jade]: Any reason behind all the extra lines added in this file demarcating each block?

File: app/views/includes/foot.jade [hunk 4]
```diff
@@ -10,20 +10,24 @@ script(type='text/javascript', src='/lib/underscore/underscore-min.js')
...
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-resource.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-cookies.js')
 
+
 //Angular UI
 script(type='text/javascript', src='/lib/angular-bootstrap/ui-bootstrap-tpls.js')
 script(type='text/javascript', src='/lib/angular-ui-utils/modules/route/route.js')
 
+
```
Comment (ooduye) [inline app/views/includes/foot.jade]: Any reason behind all the extra lines added in this file demarcating each block?

File: app/views/includes/foot.jade [hunk 5]
```diff
@@ -10,20 +10,24 @@ script(type='text/javascript', src='/lib/underscore/underscore-min.js')
 script(type='text/javascript', src='/lib/bootstrap/dist/js/bootstrap.js')
 
 //AngularJS
+
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-resource.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-cookies.js')
 
+
```
Comment (ooduye) [inline app/views/includes/foot.jade]: Any reason behind all the extra lines added in this file demarcating each block?

File: app/views/includes/foot.jade [hunk 6]
```diff
@@ -10,20 +10,24 @@ script(type='text/javascript', src='/lib/underscore/underscore-min.js')
 script(type='text/javascript', src='/lib/bootstrap/dist/js/bootstrap.js')
 
 //AngularJS
+
```
Comment (ooduye) [inline app/views/includes/foot.jade]: Any reason behind all the extra lines added in this file demarcating each block?

**Review by ooduye — COMMENTED** [review]
When you resolve all the eslint issues, test functionality again to be sure nothing is broken @andela-iamao

File: app/views/includes/foot.jade (hunk 6 above)
Comment (iAmao) [reply app/views/includes/foot.jade]: I did not touch the jade files

**Comment by iAmao** [comment]
I ran it again. No breaks.

File: app/views/includes/foot.jade (hunk 6 above)
Comment (ooduye) [reply app/views/includes/foot.jade]: But from the PR I can see that your push added these changes. If the spaces are unnecessary, get rid of them then and make your PR cleaner 😝😉

Comment (iAmao) [reply app/views/includes/foot.jade]: All right then. Will do

**Review by ooduye — APPROVED** [review]

**Pull request merged**